PurchasesCoreSwift/Public/PurchaserInfo.swift

```diff
@@ -0,0 +1,317 @@
+        description += "activeSubscriptions=\(activeSubsDescription),\n"
+        description += "nonSubscriptionTransactions=\(self.nonSubscriptionTransactions),\n"
+        description += "requestDate=\(String(describing: self.requestDate)),\n"
+        description += "firstSeen=\(String(describing: self.firstSeen)),\n"
+        description += "originalAppUserId=\(self.originalAppUserId),\n"
+        description += "entitlements=\(allEntitlementsDescription),\n"
+        description += ">"
+
+        return description
+    }
+
+    private struct SubscriberData {
+
+        let originalAppUserId: String
+        let managementURL: URL?
+        let originalApplicationVersion: String?
+        let originalPurchaseDate: Date?
+        let firstSeen: Date
+        let nonSubscriptions: [String: [[String: Any]]]
+        let entitlements: [String: Any]
+        let nonSubscriptionTransactions: [Transaction]
+        let allPurchases: [String: Any]
+        let expirationDatesByProduct: [String: Date?]
+        let purchaseDatesByProduct: [String: Date?]
+
+        init?(subscriberData: [String: Any]) {
+            let subscriptions = subscriberData["subscriptions"] as? [String: [String: Any]] ?? [String: [String: Any]]()
+
+            // Metadata
+            self.originalApplicationVersion = subscriberData["original_application_version"] as? String ?? nil
+
+            if let originalPurchaseDateString = subscriberData["original_purchase_date"] as? String {
+                self.originalPurchaseDate = PurchaserInfo.parseDate(date: originalPurchaseDateString, withDateFormatter: DateFormatter.dateFormatter)
+            } else {
+                self.originalPurchaseDate = nil
+            }
+
+            guard let firstSeenDateString = subscriberData["first_seen"] as? String,
+                  let firstSeenDate = PurchaserInfo.parseDate(date: firstSeenDateString, withDateFormatter: DateFormatter.dateFormatter) else {
+                return nil
+            }
+            self.firstSeen = firstSeenDate
+
+            guard let originalAppUserIdString = subscriberData["original_app_user_id"] as? String else {
+                return nil
+            }
+            self.originalAppUserId = originalAppUserIdString
+
+            if let managementUrlString = subscriberData["management_url"] as? String {
+                self.managementURL = parseURL(url: managementUrlString)
+            } else {
+                self.managementURL = nil
+            }
+
+            // Purchases and entitlements
+            self.nonSubscriptions = subscriberData["non_subscriptions"] as? [String: [[String: Any]]] ?? [String: [[String: Any]]]()
+            self.entitlements = subscriberData["entitlements"] as? [String: Any] ?? [String: Any]()
+            self.nonSubscriptionTransactions = TransactionsFactory().nonSubscriptionTransactions(withSubscriptionsData: nonSubscriptions, dateFormatter: DateFormatter.dateFormatter)
```

Comment:
we should probably inject the dependency, even if just as a param with a default value in the constructor, so it's easier to test, and so it's easier for us to understand dependencies between our objects

Reply:
@aboedo cool with me adding this in the dateformatter pr? since it has the convenience init